rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -137,6 +137,8 @@ def __init__(self, topic, client_id, callback, node_handle, msg_type=None, raw=F
 
     def unregister(self):
         self.node_handle.destroy_subscription(self.subscriber)
+        if self.new_subscriber:
+            self.node_handle.destroy_subscription(self.new_subscriber)
```

Comment:
This should be done inside the `with self.lock` block as the assignment of `self.new_subscriber` is also protected by the lock